libs/geoservices/covid/src/lib/pages/maps/components/time-map/time-map.component.ts

```diff
@@ -0,0 +1,293 @@
+import { Component, OnInit } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { MapboxMapService } from '@tamu-gisc/maps/mapbox';
+import { Moment } from 'moment';
+import { forkJoin } from 'rxjs';
+
+@Component({
+  selector: 'tamu-gisc-time-map',
+  templateUrl: './time-map.component.html',
+  styleUrls: ['./time-map.component.scss']
+})
+export class TimeMapComponent implements OnInit {
+  constructor(private mapService: MapboxMapService, private http: HttpClient) {}
+
+  private maxDate: string;
+  private mortalButtonToggled: boolean;
+  private stateButtonToggle: boolean;
+  private selectedDate: string;
+  private stateData: object;
+  private countyData: object;
+
+  public ngOnInit() {
+    const initObject = this;
+
+    this.mortalButtonToggled = false;
+    this.stateButtonToggle = false;
+    this.reloadData(this.mortalButtonToggled);
+
+    this.mapService.loaded.subscribe((map) => {
+      const zoomThreshold = 3;
+
+      map.addSource('county-lines', {
+        type: 'vector',
+        url: 'mapbox://gsepulveda96.countylines'
+      });
+
+      map.addSource('state-lines', {
+        type: 'vector',
+        url: 'mapbox://gsepulveda96.statelines'
+      });
+
+      map.addLayer(
+        {
+          id: 'covid-state',
+          type: 'fill',
+          source: 'state-lines',
+          'source-layer': 'state-lines',
+          layout: {
+            // make layer visible by default (change to none or visible for debugging)
+            visibility: 'none'
+          }
+        },
+        'road-minor-low'
+      );
+
+      map.addLayer(
+        {
+          id: 'covid-county',
+          type: 'fill',
+          source: 'county-lines',
+          'source-layer': 'county-lines',
+          paint: {
+            'fill-outline-color': '#ffffff'
+          }
+        },
+        'covid-state'
+      );
+
+      map.addLayer(
+        {
+          id: 'statelines',
+          type: 'line',
+          source: 'state-lines',
+          'source-layer': 'state-lines',
+          paint: {
+            'line-color': '#000000',
+            'line-width': 1
+          }
+        },
+        'state-label'
+      );
+
+      map.on('mousemove', 'covid-county', function(e) {
+        const feature = e.features[0];
+        const selectedCounty = initObject.countyData[initObject.selectedDate].filter(
```

Comment:
Use arrow functions unless you require a new scope inside the function.

This will also remove the need to bind `initObject` because arrow functions are lexically bound, meaning `this` will refer to the parents scope (component class).

Reply:
removed initObject and made the mouse moves into arrow functions